src/modules/commander/state_machine_helper.cpp

```diff
@@ -704,7 +724,19 @@ bool check_invalid_pos_nav_state(vehicle_status_s &status, bool old_failsafe, or
 				status.nav_state = vehicle_status_s::NAVIGATION_STATE_DESCEND;
 
 			} else {
-				status.nav_state = vehicle_status_s::NAVIGATION_STATE_TERMINATION;
+				const bool gnss_enabled{!estimator_selector_status.gnss_denied_instance_selected};
+				const bool propagation_time_exceeded{hrt_elapsed_time(&estimator_selector_status.last_instance_change) > 1_s};
+				const bool stale_estimator_selector_status{hrt_elapsed_time(&estimator_selector_status.timestamp) > 5_s};
```

Comment:
```suggestion
				const bool propagation_time_exceeded{hrt_elapsed_time(&estimator_selector_status.last_instance_change) > 1_s};
				const bool estimator_selector_status_stale{hrt_elapsed_time(&estimator_selector_status.timestamp) > 5_s};
```

So that the two bools have consistent naming patterns.

Reply:
The time constants are pretty arbitrary. I feel like we should not delay the failsafe by more than 1 second. Maybe the staleness should be equal though, a potential 5 second delay is a lot